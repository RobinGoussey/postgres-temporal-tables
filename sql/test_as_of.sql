CREATE EXTENSION IF NOT EXISTS psql_temporal;

CREATE TABLE test_as_of_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO test_as_of_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('test_as_of_temporal');



-- We should get the current state (1 record)
select id, name, rank from test_as_of_temporal_as_of(now());

-- This is a way to save the time, I couldn't get variables to work.
CREATE TEMP TABLE temp_time(time timestamptz);
insert into temp_time (time) values (now());

-- trigger update to update the history table
UPDATE test_as_of_temporal SET "rank" = 'Cpt.';

-- We should get the previous record state (1 record)
select id, name, rank from test_as_of_temporal_as_of((select * from temp_time));

-- We should get the current state (1 record)
select id, name, rank from test_as_of_temporal_as_of(now());



