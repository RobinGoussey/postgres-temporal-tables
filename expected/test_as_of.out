CREATE EXTENSION IF NOT EXISTS psql_temporal;
NOTICE:  extension "psql_temporal" already exists, skipping
CREATE TABLE test_as_of_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO test_as_of_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('test_as_of_temporal');
NOTICE:  trigger "test_as_of_temporal_temporal_trigger" for relation "public.test_as_of_temporal" does not exist, skipping
 convert_to_temporal 
---------------------
 t
(1 row)

-- We should get the current state (1 record)
select id, name, rank from test_as_of_temporal_as_of(now());
 id | name | rank 
----+------+------
  1 | Bram | Sgt.
(1 row)

-- This is a way to save the time, I couldn't get variables to work.
CREATE TEMP TABLE temp_time(time timestamptz);
insert into temp_time (time) values (now());
-- trigger update to update the history table
UPDATE test_as_of_temporal SET "rank" = 'Cpt.';
-- We should get the previous record state (1 record)
select id, name, rank from test_as_of_temporal_as_of((select * from temp_time));
 id | name | rank 
----+------+------
  1 | Bram | Sgt.
(1 row)

-- We should get the current state (1 record)
select id, name, rank from test_as_of_temporal_as_of(now());
 id | name | rank 
----+------+------
  1 | Bram | Cpt.
(1 row)

